src/Rendering/Text.php

```diff
@@ -234,7 +234,7 @@ private function renderVariable($data, $lang)
         }
         return $this->applyTextCase(
             StringHelper::clearApostrophes(
-                htmlspecialchars($value, ENT_HTML5)
+                htmlspecialchars(gettype($value) === "string" ? $value : implode($value), ENT_HTML5)
```

Comment:
I'd suggest to extract this to another function . For instance `valuesAsString($value)` which distinguishs `$value` not only for string but also for array and other types.

```php
 private function asString($value): string 
{
    return match(true) {
        is_array($value) => implode($value),
        is_object($value) => ...,
        is_string($value) => $value
    }
}
```

Reply:
Thanks for your suggestion. Let me try and create this function and test it against my module.

Reply:
I think I have made the issue more complex, and I would love to hear your further comments, thank you.